energy-integration/fees-collector/src/lib.rs

```diff
@@ -37,12 +46,43 @@ pub trait FeesCollector:
         let caller = self.blockchain().get_caller();
         let wrapper = FeesCollectorWrapper::new();
         let rewards = self.claim_multi(&wrapper, &caller);
+        let energy_token_id = self.locked_token_id().get();
+        let mut require_energy_update = false;
         if !rewards.is_empty() {
             self.send().direct_multi(&caller, &rewards);
+
+            let current_epoch = self.blockchain().get_block_epoch();
+            let mut energy = self.get_energy_entry(&caller);
+            for reward in &rewards {
+                if reward.token_identifier == energy_token_id && reward.token_nonce > 0 {
```

Comment:
what if we simplify and give 4 year LKMEX when the user makes a claimReward ? That way we can use the same function the farm contracts are using, that one updates the energy as well. Furthermore, we can simplify the collection of LKMEXs. Today, will make a decision about this.

Reply:
We'll be done in a new PR.